src/PasswordHasher.php

```diff
@@ -2,7 +2,7 @@
 
 interface PasswordHasher {
 
-    public function hashAndLoginPassword($password, $row);
+    public function passwordsEqual($password, $row);
```

Comment:
Pamačius tokį interface visiškai neaišku kas turėtų būti antrasis funkcijos argumentas. Siūliau variantą kaip atskirus argumentus išskirti hash, salt ir hashTimes, bet matau kad patinka turėti mažesnį skaičių argumentų (su tuo viskas ok). Tai kad ir tu patenkintas būtum, ir kolegos kodą suprastų - pasiskaitom kas yra data class, ir sukuriam ir panaudojam tokią klasę vartotojo duomenims saugoti. Taip pat pasiskaitom ir bandom visur kur įmanoma panaudoti: https://stackoverflow.com/questions/7861375/why-declare-php-variable-type-in-a-comment
Taip pat bendram išprusimui pasiskaitom, bet kolkas nenaudojame: https://thecodingmachine.io/type-hint-all-the-things

Reply:
Sveikas, sorry, kad užtrukau ši kartą,prisipažinsiu,   ne iki galo man aišku su tuo komentavimu funkcijų ir kintamųjų, tai dėl šito lauksiu pastabų, tai pat neaišku, kaip aprašyt atvejus kai gražinu statement pvz:  _return $result->num_rows > 0_  - čia parašiau kad boolean, nes realiai tai tikrinam TRUE or FALSE , taip pat atvejų kai nieko negražinam - šiuo atveju parašiau NULL , bet gal čia nereikėjo nieko ir nerašyt ?